tests/pipelines/audio_diffusion/test_audio_diffusion.py

```diff
@@ -0,0 +1,159 @@
+            block_out_channels=(128, 128),
+            down_block_types=("AttnDownBlock2D", "DownBlock2D"),
+            up_block_types=("UpBlock2D", "AttnUpBlock2D"),
+        )
+        return vqvae, unet
+
+    def test_audio_diffusion(self):
+        device = "cpu"  # ensure determinism for the device-dependent torch.Generator
+        mel = Mel()
+
+        scheduler = DDPMScheduler()
+        pipe = AudioDiffusionPipeline(unet=self.dummy_unet, scheduler=scheduler)
+        pipe = pipe.to(device)
+        pipe.set_progress_bar_config(disable=None)
+
+        generator = torch.Generator(device=device).manual_seed(42)
+        output = pipe(mel=mel, generator=generator, steps=4)
+        audio = output.audios[0]
+        image = output.images[0]
+
+        generator = torch.Generator(device=device).manual_seed(42)
+        output = pipe(mel=mel, generator=generator, steps=4, return_dict=False)
+        image_from_tuple = output[0][0]
+
+        assert audio.shape == (1, (self.dummy_unet.sample_size[1] - 1) * mel.hop_length)
+        assert image.height == self.dummy_unet.sample_size[0] and image.width == self.dummy_unet.sample_size[1]
+        image_slice = np.frombuffer(image.tobytes(), dtype="uint8")[:10]
+        image_from_tuple_slice = np.frombuffer(image_from_tuple.tobytes(), dtype="uint8")[:10]
+        expected_slice = np.array([255, 255, 255, 0, 181, 0, 124, 0, 15, 255])
+        assert np.abs(image_slice.flatten() - expected_slice).max() == 0
+        assert np.abs(image_from_tuple_slice.flatten() - expected_slice).max() == 0
+
+        scheduler = DDIMScheduler()
+        dummy_vqvae_and_unet = self.dummy_vqvae_and_unet
+        pipe = LatentAudioDiffusionPipeline(
+            vqvae=self.dummy_vqvae_and_unet[0], unet=dummy_vqvae_and_unet[1], scheduler=scheduler
+        )
+        pipe = pipe.to(device)
+        pipe.set_progress_bar_config(disable=None)
+
+        np.random.seed(0)
+        raw_audio = np.random.uniform(-1, 1, ((dummy_vqvae_and_unet[0].sample_size[1] - 1) * mel.hop_length,))
+        generator = torch.Generator(device=device).manual_seed(42)
+        output = pipe(mel=mel, raw_audio=raw_audio, generator=generator, start_step=5, steps=10)
+        image = output.images[0]
+
+        assert (
+            image.height == self.dummy_vqvae_and_unet[0].sample_size[0]
+            and image.width == self.dummy_vqvae_and_unet[0].sample_size[1]
+        )
+        image_slice = np.frombuffer(image.tobytes(), dtype="uint8")[:10]
+        expected_slice = np.array([120, 117, 110, 109, 138, 167, 138, 148, 132, 121])
+        assert np.abs(image_slice.flatten() - expected_slice).max() == 0
+
+
+@slow
+@require_torch_gpu
+class PipelineIntegrationTests(unittest.TestCase):
+    def tearDown(self):
+        # clean up the VRAM after each test
+        super().tearDown()
+        gc.collect()
+        torch.cuda.empty_cache()
+
+    def test_audio_diffusion(self):
+        device = torch_device
+
+        mel = Mel()
```

Comment:
Should we maybe create this automatically directly in the pipeline? It might be a bit more user-friendly?